utils/slices.go

```
@@ -39,3 +42,26 @@ func AnyOf[T comparable](e T, values ...T) bool {
 	}
 	return false
 }
+
+// Unique returns a new slice with duplicates removed
+func Unique[T comparable](slice []T) []T {
+	if len(slice) == 0 {
+		return slice
+	}
+
+	// do not support unique on pointer types, just return the slice as it is
+	if reflect.TypeOf(slice[0]).Kind() == reflect.Ptr {
+		return slice
+	}
```

I think we should add this to the description of `Unique`. Also, I think it is Ok to panic here since it is meant for internal use instead of silently not doing anything.

---

Added